### ISSUE TITLE
Random newly added text translations disappear in the FF Mac App

### ISSUE THREAD

**Issue by khomya** [FlutterFlow/flutterflow-issues#2346]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I add new elements to the application (text, button) and add translations to other languages (I have 5 languages in my project).
when I RUN the project or Deploy to Mobile -> I get application pages with missing texts in languages other than the main one.
In this case, the texts disappear randomly. I couldn't trace the pattern. Some texts need to be edited FIVE times in order to preserve the translations. Some of them need to be edited only once.

Application version 4.1.26
This bug appeared just a week and a half ago.

If I edit the project via the web, there are no problems.

### Expected Behavior

Text translations should not be lost

### Steps to Reproduce

1. Add text element on page
2. Add translations to text via "hands", via copy-paste or via Google Translation button
3. RUN or Deploy to Mobile
4. Open app - and see that translations is empty

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4wkfHlsIZ2xOFY+KXqdcFv+SAWCDp/aJ0/lu4aRis9GYjyO5YUacnRPxRoQtOhSmxXKEWJhlsypdrVkvLXCMECNRKsc6plwaoNEDzJQk2/aoTXO7jhQUB9I+FPG1Sd37W0kxB+BNdZdE443UOhevOQdCzsQJP6Zwh9f6PHaOI=

### Context

Text translations should not be lost

### Visual documentation

![Screenshot 2024-02-15 at 12 32 16](https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/e09d3a2e-1bd4-40b0-8ce3-a27db4d31481/7dc8accc-2a14-4f26-8cf6-b3c7390c70d4?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS9lMDlkM2EyZS0xYmQ0LTQwYjAtOGNlMy1hMjdkYjRkMzE0ODEvN2RjOGFjY2MtMmExNC00ZjI2LThjZjYtYjNjNzM5MGM3MGQ0IiwiaWF0IjoxNzA3OTkwMTg0LCJleHAiOjMzMjc4NTUwMTg0fQ.9hyOgodVKRHsFh4KwnWO2OeO5sK2K82mj_k2kVbyWJ4 "=1239x")

### Additional Info

*No response*

### Environment

```markdown
- FlutterFlow version: 4.1.26
- Platform: Mac
- Browser name and version: Chrome 121.0.6167.160
- Operating system and version affected: MacOS 13.6.4 (22G513)
```

**Comment by SitnAlex** [comment]
Have the same issue in the web, texts just randomly disappear, sometimes overwritten

I've noticed this bug more than two week ago, but it was minor issues, maybe one or two text widgets/buttons in the whole project. But now it's almost impossible to work in the project because almost every text fails to work right.

Met this bug in google chrome and safari

Here are the examples:
<img width="330" alt="english_texts" src="https://github.com/FlutterFlow/flutterflow-issues/assets/78790768/56009613-b4b2-4253-a4af-28dbb6c9e48c">
<img width="330" alt="ukranian_texts" src="https://github.com/FlutterFlow/flutterflow-issues/assets/78790768/0c30decd-b982-4d6b-a849-f63ef31aeb42">

https://github.com/FlutterFlow/flutterflow-issues/assets/78790768/01aae5fa-0f7e-4818-8f51-91ccb7c7a17c

**Comment by khomya** [comment]
Another example of bug.

I just made a duplicate of component.
And got this.
<img width="337" alt="Screenshot 2024-02-16 at 15 11 19" src="https://github.com/FlutterFlow/flutterflow-issues/assets/2369704/4ea660c9-88fa-4dd5-8ca3-79d71251f871">

<img width="336" alt="Screenshot 2024-02-16 at 15 11 30" src="https://github.com/FlutterFlow/flutterflow-issues/assets/2369704/582006c5-ae11-40cb-8f10-0527513d7c3a">

**Comment by SitnAlex** [comment]
Hey @hariprasadms @takashiff, do you have any news on this issue?

**Comment by AlexFox63** [comment]
Hey, @hariprasadms @takashiff guys? Any updates? We have a really big project which was launched on December and this feature affects us. Please fix it asap

**Comment by SitnAlex** [comment]
Hey, @hariprasadms @takashiff did you forget about this issue? Give us any info on that.
We have a really big app, more than 5000 users! And we need to postpone new releases because of this issue!

**Comment by SitnAlex** [comment]
@hariprasadms @takashiff any news?

**Comment by AlexFox63** [comment]
@hariprasadms @takashiff any updates?

**Comment by AlexFox63** [comment]
@sbis04 guys it's a serious problem. Do you have any updates regarding this issue? It's torture to edit projects with the localizations. If it continues we'll be supposed to migrate from the FlutterFlow to the usual Flutter development.

**Comment by SitnAlex** [comment]
@hariprasadms @takashiff any updates?

**Comment by sabatage3** [comment]
I've been dealing with this issue for months now. Recent translations keep disappearing on their own. I've had to add the same translation up to three times just to keep it. And it's not related to the Mac version; the problem lies with FlutterFlow and occurs on any platform. It's incredibly frustrating to have to redo translations manually over and over again. The worst part is there's no way to know that the translations have been deleted unless you stumble upon empty text on the platform. This is disastrous for a project that's already in production. Please @takashiff  @hariprasadms  we need help!!

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/15147720/345de98b-5cf3-4ea8-a28b-8160c92b505f)

**Comment by khomya** [comment]
Well. Its veeeery bad that this bug is still NOT fixed.

I wanted to add 5 more languages to my app in march, but with current FF state is just impossible :(

**Comment by ryutosh** [comment]
I mainly take the method from TextWidget where I click on the globe icon and type in my translations.
And FF breaks my translations so often that there is no way the FF team is unaware of this problem.
I don't know how to reproduce this as I never work only on translations, but it seems that when I add or duplicate a TextWidget and add a translation, unrelated translations seems also broken.

**Comment by leighajarett** [comment]
So sorry that you're all dealing with this! We're taking a closer look now. Thanks so much for filing

**Comment by takashiff** [comment]
Hi all, I apologize for the delayed response. Does this data loss occur directly after making changes to the translations, or have there been cases where after an extended period of time, translations have been lost? Thanks

**Comment by sabatage3** [comment]
> Hi all, I apologize for the delayed response. Does this data loss occur directly after making changes to the translations, or have there been cases where after an extended period of time, translations have been lost? Thanks

Hi @takashiff it happens mostly with recent translations, sometimes i find myself retranslating the same page or component up to 4 times, flutterflow deletes lot of translations, however, i remember seeing holes in translations i did in the past

**Comment by AlexFox63** [comment]
> Hi all, I apologize for the delayed response. Does this data loss occur directly after making changes to the translations, or have there been cases where after an extended period of time, translations have been lost? Thanks

Also guys I think it will be useful to get rid of the current localization mechanism functionality because during application development translations can be repeated and there is no point in creating a translation with a new randomly generated Id. Also, refactoring code with translations is not an easy task because you need to find those randomly generated identifiers and put those translations into widgets. Please make localization like in usual Flutter development, where you need to customize translation Ids yourself and reuse the same Id throughout the whole project or even between projects.

**Comment by takashiff** [comment]
@sabatage3 Thanks for the additional info! Sometimes it takes a few seconds for the translations to be saved onto our backend, so this may be what is causing the issue. I would advise waiting a few seconds after running the translations before switching to another tab or part of the project. Please let me know if this resolves your issue

**Comment by takashiff** [comment]
@AlexFox63 Thanks for the feedback! We are always looking for ways to improve our product, and I appreciate you taking the time to write out a detailed improvement to our localization feature. If you'd like, I can pass this along internally so that it may be reviewed by our development team.

**Comment by SitnAlex** [comment]
> @sabatage3 Thanks for the additional info! Sometimes it takes a few seconds for the translations to be saved onto our backend, so this may be what is causing the issue. I would advise waiting a few seconds after running the translations before switching to another tab or part of the project. Please let me know if this resolves your issue

Have you seen the attached video? There's a text overwritten in realtime. And yes, I've tried waiting for a minute after changing text widget and it doesn't help.

**Comment by takashiff** [comment]
Hi @SitnAlex, the bug displayed in the video appears to be different from the bug originally described. Could you provide the bug report code to a widget in your project? This will let me take a closer look at the behavior showcased in the video, as I am currently unable to reproduce it in my own test projects. Thanks

**Comment by takashiff** [comment]
@SitnAlex Thanks for the additional information! When you encountered this error, was anyone else working on the project at the same time as you? I suspect this may be a bug with our collaboration system

**Comment by SitnAlex** [comment]
> @SitnAlex Thanks for the additional information! When you encountered this error, was anyone else working on the project at the same time as you? I suspect this may be a bug with our collaboration system

No, nobody was working except me

**Comment by takashiff** [comment]
@SitnAlex Got it, I appreciate the quick response

**Comment by takashiff** [comment]
Hi, just wanted to provide an update. I have finally been able to reproduce the issue, and we are currently working on a fix. I appreciate the prompt communication and responsiveness. Thank you all for your continued patience and support.

**Comment by ready4sea** [comment]
May I also add that not only translations but also the tick mark of the "Fixed" column also disappears randomly. There should be a global setting for this one anyhow.

**Comment by ryutosh** [comment]
> Hi, just wanted to provide an update. I have finally been able to reproduce the issue, and we are currently working on a fix. I appreciate the prompt communication and responsiveness. Thank you all for your continued patience and support.

@takashiff 
You may have already identified the cause of the problem, but I am adding a point that I noticed today.
I also hope that this will help identify the problem, as there is not always just one factor that causes a system bug.

# Problem
I wanted to radically revise the layout of the TabBar on a page, so I deleted the entire TabBarWidget that I had been developing.

Then I added a new TabBar and to my surprise, the translated text for Tab was already set.
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/71914270/377a1be4-04d9-4be0-8bab-9174f8fdcf59)

My expectation was that the translated text of the deleted Widget would not be remembered, and that the translated text of the newly added Widget would be unset. Perhaps, are FF remembering the translation sentences per each type of widgets or similar not per each instance of widgets?

**Comment by khomya** [comment]
I have a text widget. and i created translations for it, i think more that 20 times.
And it newer saves.

Pls look into it, Maybe it will help catch this bug:

IT4wkfHlsIZ2xOFY+KXqdcFv+SAWCDp/aJ0/lu4aRis9GYjyO5YUac/RdVtoZMfjTQk6KEWIuFEE/sr6hYSWIewnG06VfKp6z6lQTQ7/Rl6/aoTXO7jhQUB9I+FPG1Sd37W0kxB+BNdZdE443UOhevOQdCzsQJP6Zwh9f6PHaOI=

**Comment by Thesergiolg99** [comment]
Having the same problem here on my project, every day new traslations are missing again, randomly texts.

**Comment by leighajarett** [comment]
We are working on a fix for this and hope this will be resolved ASAP!

**Comment by ready4sea** [comment]
Thank you, was about to ask you of anything was underway because this is terrible time wasting issue. Solving it would be a huge time "saving", i.e. not having to re-input translations multiple times.

**Comment by leighajarett** [comment]
Understood, we're working on getting the fix merged!

**Comment by SitnAlex** [comment]
Hey @leighajarett can you tell us approximately when this will be fixed?

**Comment by leighajarett** [comment]
The fix is still in engineering review but is a bit more complex than we originally thought. Hopefully will be out in the next week or two!

**Comment by ready4sea** [comment]
Wondering if the status of this ticket has moved on. Indeed, it is still as problematic as ever to input translations. Even when inputting text multiple times, it is sometimes nearly impossible to have FF save the texts. When downloading the source code, the `internationalization.dart` file comes up desperately empty except for the default language.

I don't know if it's a false impression, but I feel like it's particularly painful on widgets that have been duplicated or copied from elsewhere. In any case, it makes releasing multi-language versions extremely painful and looking forward to a prompt resolution !

**Comment by AlexFox63** [comment]
> The fix is still in engineering review but is a bit more complex than we originally thought. Hopefully will be out in the next week or two!

Can we wait for the update on this week with the fix?

**Comment by SitnAlex** [comment]
Hey there guys! @hariprasadms @leighajarett @takashiff

Just wanted you to know that once again we successfully postponed our release due to the fact that some of the translations have disappeared.

We are planning a huge update at the end of the month with a lot of tasks to do, but we're still have to waste time solving the problem with translations...

**Comment by leighajarett** [comment]
Sorry that you're dealing with this! We're trying to get a fix out ASAP

**Comment by Thesergiolg99** [comment]
Good morning everyone,

Issues still here, everyday i start working on my project, new traslations have dissappeared or even changed to a completely different traslation.

I think this bug is critical and should be on the main scope as it delaying every project.

Hope to have news on this soon.
Regards.

**Comment by ready4sea** [comment]
Completely agree @Thesergiolg99,
I lose about 30% of every day's work having to redo stuff that was already done, fixing disappeared translations, or simply waiting for the translations to be saved and checking that they have been taken into account.
Perhaps monolingual users in the US can't see the value of this, but clearly this is very critical pretty much everywhere else.

**Comment by leighajarett** [comment]
I'm so sorry, I know it's frustrating - it's a high priority for us and we're actively working on it

**Comment by Nic61** [comment]
I just want to hop in and confirm it happens to me as well. Both on web and mac app. Both on individual elements and via the languages settings tab of FF.
Hope to get some updates on this.
Thanks FF team.

**Comment by Nic61** [comment]
@leighajarett do you know if there's a way to help this not happen? E.g. once clicked translate wait for 10seconds for the saving to occur. Or click in another field to trigger the saving. Or something like that.

**Comment by leighajarett** [comment]
Hi there! We just merged a fix and it should be included in next week's release. I am not sure how to help this not happen, I would need to take a closer look

**Comment by YassinElsherif** [comment]
@leighajarett is it available already? I just has the same issue in Version 4.1.45 (4.1.45)

**Comment by mattia-reguzzoni-kingsrl** [comment]
I'm having a similar issue with the RichText widget, if I try to set a translation for the RichTextSpan it doesn't get saved.

Also the automatic translation with google translate doesn't seem to work, leaving empty all the remaining boxes.
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/92975550/a7bb39c4-3d69-46e9-8b09-deb514a2d9b2)

**Comment by leighajarett** [comment]
It's currently in beta and show be available for everyone after today's release

**Comment by ready4sea** [comment]
Same issue still happening with v4.1.47+, whether on the web or using the macOS app.

**Comment by leighajarett** [comment]
@SitnAlex we are not able to reproduce it. Can you please send us the exact steps to reproduce from a blank project?

**Comment by leighajarett** [comment]
@AlexFox63 thanks for the feedback. We're definitely hoping to rethink the way we do localization soon

**Comment by SitnAlex** [comment]
> @SitnAlex we are not able to reproduce it. Can you please send us the exact steps to reproduce from a blank project?

Just add a text, enter translations, do anything in the editor for 10-15 minutes. After 10-15 minutes translations in new text widgets may disappear. Maybe you would need to spend up to 30 minutes, maybe exit the app, maybe launch run mode. I just don't know when it happens, it's hard to understand what specific action you need to do to reproduce it. At some point when you go back to the page with new text widget you'll notice that translations disappear.

Please use another language while doing something in ff editor (For example if the main language is english, switch to another language, that'll help you find disappeared texts, because translations disappear in every language except main language.

Try adding text widget and duplicating it, then editing translations. Try duplicating already existing text widgets and editing translations. Just add a lot of text widgets, like 20-30 and you'll likely reproduce the bug

**Comment by Nic61** [comment]
Just to chip in. Still happens to me as well. Sometimes I click the Google Translate. Then leave the project and close FF mac. Open up again and it wasn't saved.

**Comment by almantas88** [comment]
Happening to me as well. I am working with four languages and this issue is really frustrating. PLEASE FF TEAM HAVE A LOOK AT THIS

**Comment by leighajarett** [comment]
We are actively looking at this

**Comment by MxMartin** [comment]
This arguably is the most annoying issue in FlutterFlow and keeps me from using FlutterFlow for additional projects. I agree that translations in FlutterFlow should be reworked completely, to also allow for easier placeholders, plurals and selects.

That said. I managed to record an instance where the translation disappears. It happens in the moment when I press the "globe" icon to edit a multi language string. At that moment, other translations visible on the same page disappear. I can provide a recording if you let me know here to send it.

**Comment by khomya** [comment]
Are there any news about this bug?
THREE months have passed since I posted this issue.

I cant develop new functions and I cant add new languages for my app because of this mega-bug! :(

**Comment by leighajarett** [comment]
So sorry about that 🙁 I know it's super frustrating. This has been a tough one to fix but we're actively looking into it!

**Comment by Nic61** [comment]
The multi language support is one of the best features of FF, however it really is tough to work with this bug.
Hope we get a fix soon. All the best to the FF team.

**Comment by Nic61** [comment]
Did anything get shipped in the latest mac update?

**Comment by khomya** [comment]
No, The bug is still there. :(

**Comment by leighajarett** [comment]
We think we may have a fix under review, we hope to get this out ASAP. Once again we're really sorry for the frustration this has caused

**Comment by Nic61** [comment]
Awesome! Is the 'completed' because the fix has been tested successfully? When will it be shipped?

**Comment by leighajarett** [comment]
Yes the fix has been merged! Should be live next release (next Thursday morning EST)

**Comment by khomya** [comment]
Well... I installed Mac App FlutterFlow 4.1.54 - and everything got WORSE :(

Before - it took 1-2 times to properly save the translations. Now - its 5-6 times.

Unbelievable. Its almost impossible to add something to the app now :(

**Comment by almantas88** [comment]
Seems like the issue still persist... Tried to add a language translation in the languages tab, refreshed the app and the translation didn't save. I have waited some time, saved the project (manually clicking the cmd + s).

 When I added in the editor it seems to be saved.

**Comment by leighajarett** [comment]
The fix for the issue is live in our beta environment and will be included in next week's production release. We shifted the dates to Tuesday so it will be released a few days later than I originally mentioned, sorry about that!

**Comment by khomya** [comment]
Well! I tried 
FlutterFlow v4.1.59 released June 13, 2024
Flutter version is 3.19.1

The bug is still there.

I just added a Button to the project.
And this is what I get:

<img width="804" alt="Screenshot 2024-06-13 at 10 36 10" src="https://github.com/FlutterFlow/flutterflow-issues/assets/2369704/1cd2360f-adeb-4c2f-be54-c8ee0054328c">

In default language its "Button".
In all other languages its "Add court" (I have such button! AND if I change translations to the NEW button - the text in "Add court" court button will change too! Its total mess!)

**Comment by SitnAlex** [comment]
Same here, just added a new text to check and all the translations changed to random
<img width="928" alt="Screenshot 2024-06-13 at 11 50 46" src="https://github.com/FlutterFlow/flutterflow-issues/assets/78790768/f2e780eb-b1fb-42ad-bc6f-21b509f5cf7e">

**Comment by leighajarett** [comment]
Oh no 🙁 Okay taking a look!! So sorry about that

**Comment by AlexFox63** [comment]
@leighajarett @hariprasadms Guys let's face it :) You need completely refactor your logic with internationalization and make it by using standard Flutter tools https://docs.flutter.dev/ui/accessibility-and-internationalization/internationalization#introduction-to-localizations-in-flutter with .arb files. It works well and for existing projects, you need to create a script to parse all translations and build an arb files from them to make the process without any pain.

**Comment by khomya** [comment]
Any news?

**Comment by khomya** [comment]
Its 4th month now - that I cant add new big functions to my app - because its impossible to create translations to different languages.

**Comment by leighajarett** [comment]
I understand, we thought we had fixed this issue but we are re-looking into this

**Comment by fettespferd** [comment]
Yes this problem is also effecting us and it is severe, because translations are being deleted randomly. For users in secondary languages the app is not usable at all because certain clickable links and texts are just not there. Pls prioritize and fix.

**Comment by fernandaflx** [comment]
I have the same problem, the translations disappear randomly. 
The application I’m building is multi-language and this problem is making a big impact on project delivery.

**Comment by aruana-dev** [comment]
@leighajarett: any update on this guys?
over 4 months now .. thats ridiculous .. why even announce new features when something basic like translation doesn't work .. I cannot ship my app as long as this has not been resolved :-(

**Comment by ryutosh** [comment]
It seems that not only "Newly added" as in the title of this Issue, but also "Already present" widget translations will be broken. (ver 4.1.61+)
> I cannot ship my app as long as this has not been resolved :-(

Yes, totally agree.
Every time before your app release, all translations must be checked for corruption, including the widgets that not touched.
Hence, this problem has more impact on projects that are already running than on new projects. We should be cautious.

> This arguably is the most annoying issue in FlutterFlow and keeps me from using FlutterFlow for additional projects. I agree that translations in FlutterFlow should be reworked completely, to also allow for easier placeholders, plurals and selects.

I agree that FF's translations don't understand secondary language users, and since FF has many Indian developers, FF might want to heed their advice better. When I considered CMS products for my company, I found that products from countries whose first language is English often underestimate the difficulties of multilingualism. (Many products thought that dynamic automatic translation plugin would enough)

However, other than this Issue, there are many things that can be avoided by creating functions on the FF user side. But fixed texts like the attached cannot be prevented by us and should be avoided.

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/71914270/05a2b362-a2e5-468b-83f3-c33730227b6d)

**Comment by YassinElsherif** [comment]
any updates about when you expect this to be solved?

**Comment by lleo132** [comment]
Same problem here, I noticed that using the Mac or Windows version makes this situation worse, I have been losing translations in my app from one day to the next.

**Comment by YassinElsherif** [comment]
@leighajarett I see that the issue is now closed, is it already fixed?

**Comment by leighajarett** [comment]
Hi! Yes we have a fix merged for this issue, it should be available in the next week or so - going through some testing now

**Comment by joaocordeiro-dotted** [comment]
Please, it's so annoying to develop stuff with this problem. It's happening all the time with the text widgets inside a component. It's been 4 months guys, omg....

**Comment by joaocordeiro-dotted** [comment]
Guys...look at this video I made of the problem. How are we supposed to work with a bug like that?

https://www.loom.com/share/dc7483a02b9a4110a0c39c64318fe5c5?sid=875a6c5d-a4ee-4612-9171-63a21cc71a07

Bug report code: ITFXlMmB2892odxE7rrIK8ZvnjwwFjsaU40V0tdTSQ43GJ/mBIkAP+r+VHBYOOGYT3NEIlWagmEz0M7KveLyUO0qMk6BeINo+aVyExbhRmGjRcWOPaqeXHxCP+ZQGn6j0KaFhxBTOvheV0Zmw1r/cq3qNleeY8aSfxBlZ7vfcPo=

**Comment by joaocordeiro-dotted** [comment]
@leighajarett 
Recordered a step by step video in a blank project to help you guys reproduce.

The bug is insane...I'm doing it for the second time in a blank new project, just to show you guys the pain we have in a bigger project.

I just made this:

1. Created a component with a component parameter of type string
2. Inserted a text widget on the component that receives the parameter we created
3. On a new page, I inserted the same component 2 times
4. Allowed translations on both components on "text" parameter
5. Typed different texts in both components

The result is that it mixes the text between the two components.

Bug report code: ITEez/Hqx892pbwB0aX+acJsqwMwGkQ5a7sviu5+FBQaI+LnE4kAPOD8X1htWOqHeFZAJGeIkkYC0MLWi/HpGO1eBzmtGoB/zqhfUxDgJXKQVq63Bsywen9ARMNWf2K10ba3hBFTKsphLnwC7Tiqf9mQY3qCf9qOYwx5e6fDbOY=

video: https://www.loom.com/share/e253f90624b444dc87605ff83cb154d7?sid=1a945929-b899-4401-a2b4-93979c5d50a0

**Comment by rafaelosiris** [comment]
Hey guys we also have the same issue. Please solve this ASAP. This have a huge impact in our development and release's.

**Comment by kazz91** [comment]
I have the exact same bug.. its so annoying.. FF team, fix it please  =/

**Comment by joaocordeiro-dotted** [comment]
It's still happening. Less than before, but happening. It's so annoying guys.

**Comment by kazz91** [comment]
I just created a page with a form, left the page, and came back, and look how it turned out... Honestly, it's impossible to work like this. I'm already looking for an alternative to FlutterFlow, unfortunately.

![cc72930a-07ea-4a46-83dc-52939760d2a6](https://github.com/user-attachments/assets/85e1837d-0848-4f14-be9a-aee55f79cf19)

**Comment by kazz91** [comment]
I recorded the problem a little while ago. It seems like it is not respecting which field is being altered, so I am always changing another field instead of only changing the selected one:

https://github.com/user-attachments/assets/169b79db-d930-47e4-8d20-daabff4aba11

**Comment by leighajarett** [comment]
We're actively working on this, thanks for all the information - so sorry for headache 🙁

**Comment by lleo132** [comment]
There is definitely no way to use Flutterflow in my company's project, the multilingual nature simply makes it impossible for our team to develop.

**Comment by joaocordeiro-dotted** [comment]
I love Flutterflow, but this kind of bugs play against the good things...

**Comment by leighajarett** [comment]
Hi all - okay we have a fix for this currently in beta that should go into production tomorrow. Please test it out after the release tomorrow and let me know if you are still seeing any issues

**Comment by ryutosh** [comment]
> Hi all - okay we have a fix for this currently in beta that should go into production tomorrow. Please test it out after the release tomorrow and let me know if you are still seeing any issues

@leighajarett 
Is the FlutterFlow team testing it? (passed?)

**Comment by fettespferd** [comment]
I tested the new version today and added/generated translations. But it still seems to be flawed for me. Some of the translation are not being displayed event though they are visible in the editor. Might be a slightly different problem, but still super annoying.

**Comment by leighajarett** [comment]
Hmm that sounds like a separate issue, can you file a different issue with specific details  (like a video recording) and we can take a look?

**Comment by leighajarett** [comment]
But agreed that does sound annoying, we'll take a look when we have more details, thank you for flagging!

**Comment by khomya** [comment]
For me EVERYTHING is still as it was! :(

FF v.4.1.76 Mac Application

The translations randomly dissappear.
The text in Editor displayed - its some times not the text that is set in the translations.

Dissapointment! :(

**Comment by khomya** [comment]
<img width="1345" alt="Screenshot 2024-07-25 at 13 33 55" src="https://github.com/user-attachments/assets/39d8c152-f944-4299-9e84-fc1668d11c11">
Here is the example.

All 5 text disappear. I translate them once again (I think for the 5th time is a row!).
And now - 2 texts are missing completly from English.
And 3 texts are displayed, BUT there are no text in English translation.

**Comment by szymonpodwyszynski** [comment]
Translations are still bugged, after copying and pasting elements with text, and after waiting some time or refreshing the page couple of times, the translations disappear randomly!
Not fixed

**Comment by leighajarett** [comment]
So sorry about this, we are working on it 🙁

**Comment by leighajarett** [comment]
Is anyone able to reproduce this consistently?

**Comment by joaocordeiro-dotted** [comment]
@leighajarett  yes, I am. Just updated my desktop app (MacOS) with the FF version v4.1.79. When I create a text widget and change the translations texts, it still not saving the "portuguese" text. Minutes after the translation field is empty.

**Comment by khomya** [comment]
> @leighajarett yes, I am. Just updated my desktop app (MacOS) with the FF version v4.1.79. When I create a text widget and change the translations texts, it still not saving the "portuguese" text. Minutes after the translation field is empty.

I have exactly the same problem!
Even three cases:
1. create a widget (text, dropdown, etc.), generate translations automatically or enter them manually - the next time you open the page in a few minutes (or if you close the FF application and then open it), the translations disappear
2. If you make a duplicate of widget or copy-paste widget.
The new widget also loses translations after a few minutes or the next time you open the FF application.
3. Randomly, some widgets in the application suddenly show a COMPLETELY different text. and in half of the cases there is no way to fix it. even after correcting the translation 20 times, it still shows the wrong one. and such cases are solved only by deleting the widget and creating a new one.

On the plus side, there are no problems when creating a new widget and creating translations for it. transfers do not disappear in this case.

**Comment by ryutosh** [comment]
1. Select Text Widget to change translation in UI builder screen 
2. Click "World" Icon in Text Field
3. Change like bellow
   - English: AAA --> XXX
   - Japanese: BBB --> YYY
4. Focus out and wait few minutes (My last case, 74secs)
5. Both XXX and YYY translations back to AAA and BBB.

Note:
This issue also happen if I click "Back" at step 4 and I can notice because of text change in UI screen.

Please forgive the lack of captures as it pertains to our product information.
The platform is Mac OS and the version is 4.1.82.

It is very frustrating that these quick to find and data loss type bugs have repeatedly skipped over FF reviews, then losting our lot of time and results.

I am very grateful for the time efficiency and innovation FF provide (over and above what we have lost). However, your product management is taking so much away from us and I sincerely hope FF can improve.

**Comment by ryutosh** [comment]
Let me post this as it may be a hint for FF team.
If this is the wrong consideration, please feel free to dismiss it.

---
Text widgets have a variety of events, making state management difficult.

Having an "Apply" button in the translation screen and changing the specification to reflecting texts when the "Apply" button is pressed can simplify things.
![image](https://github.com/user-attachments/assets/51e6f360-a78a-41d7-91f7-fa07f82cda69)

I am not mean this change removes the cause of the bug, but making things less complicated than needed contributes to the overall quality of the product, and may make easier to fix the problem.

Since we, FF users, basically work with mouse and keyboard on large layouts screen, so this change will not affect usability.

**Comment by leighajarett** [comment]
Hi there! Oof I know this issue has been super painful for you all. We've been working hard to fix the root issue - and we think we have it solve now. Please let me know by commenting on this issue if you are still seeing the problem

**Comment by khomya** [comment]
FlutterFlow Mac 4.1.82
The bug is still there.
Translations dissapear as usual.

**Comment by leighajarett** [comment]
Oh noooo 🙁 the bug that never ends, ok we're back on it!!

**Comment by MxMartin** [comment]
I confirm the bug still exists :/

It now seems to exclusively occur after copying widgets or pages (i.e. after copying several pages, all translations on the copied pages disappeared consistently, while components used in these pages were not affected).

An aspect that might play a role that as far as I can see has not been addressed, yet, here, is that in my case and as far as I can tell in all the cases where screenshots were posted here, the primary language of the app is different than the display language.

**Comment by khomya** [comment]
Why its closed? The bug is still there...

**Comment by leighajarett** [comment]
Sorry for the misunderstanding, we have these issues hooked up to our continuous integration system - theyre closed automatically when a fix has been merged in. It's currently undergoing some beta testing and then will be available in production

**Comment by SitnAlex** [comment]
Guys, today the bug turns 6 months old (since creating this issue). 

6 months since we can't add new features to our apps without pain and suffering. 6 months we have to explain to our users why updates are delayed while flutterflow has no problem adding new "importatnt" features.

Just wanted to notice that the bug is older than:
* Shadows for Text Widget
* Implicitly Animated Text Styles
* Gradient Color by Variable
* Redesigned Color Picker

Do you guys remember how old color picker looked?

**Comment by leighajarett** [comment]
I totally understand the frustration 🙁 <br><br>We've been working really hard to solve this issue, it's been a tricky one. As mentioned above, we have a fix out going through testing - hopefully this solves it once and for all!

**Comment by ryutosh** [comment]
The issue still present in 4.1.83 (Mac).

How we can know which version number "merged" pointed about?
Does FlutterFlow have release note like this?
![image](https://github.com/user-attachments/assets/4399976c-a5ad-42c7-bd43-c7c561bc4b92)

**Comment by celiostat** [comment]
Still having the issue here 
FF desktop app : v4.1.85 on windows.

**Comment by ryutosh** [comment]
Still having the issue.
I have seen this occur with the hint property of a text field.
I have not checked any other widgets or fields.
( please imagine trying so hard and trying so many that they all may get lost)

FF Mac: v4.1.85

**Comment by ryutosh** [comment]
It's a gut feeling, but I think it has something in common with this issue that frustrates me every time.

https://github.com/user-attachments/assets/43a60bb5-e75b-46d2-a4f9-3356e4fef843

**Comment by khomya** [comment]
> It's a gut feeling, but I think it has something in common with this issue that frustrates me every time.
> 
>  Screen.Recording.2024-09-04.at.9.39.55.mov

I have the same bug for last 2 weeks!

**Comment by sabatage3** [comment]
> It's a gut feeling, but I think it has something in common with this issue that frustrates me every time.
> 
>  Screen.Recording.2024-09-04.at.9.39.55.mov

Yeah, this is happening to me too, also when I create default values, the disappear too

**Comment by ryutosh** [comment]
It has been observed that the issue continues to be present in version 4.1.87 Mac.
(After changing the title of a page created from a newly blank, the default language en and the additional language ja disappeared)

@leighajarett 
Anyway, this issue is closed. Is it correct to report it here?

As a developer, I understand that FF is taking a long time as this issue is complicated.
This issue is destructive of developer effort, and I believe everyone wants to see a release that works correctly, even if it takes a long time to get it right.

However, for those users who have been unable to release for a long time because of this Issue, can you advise on interim workarounds?

One way I am thinking is to modify the kTranslationsMap(*1) that FlutterFlow generates when merging into the develop branch(*2), without changing the keys of the map.
Because the keys seem to be immutable against code generations, I'm thinking it's a reasonable idea for users who are really having trouble.

*1 .. /lib/flutter_flow/internationalization.dart
*2 .. https://docs.flutterflow.io/exporting/push-to-github/#manage-custom-code-on-github

**Comment by leighajarett** [comment]
Hi all, so sorry you're still experiencing an issue here 🙁 <br><br>Can you do me a favor and try adding a language and removing it? Also can you please send over the bug report code for the project where you are experiencing the issue, along with a screen recording of the behavior in the running app. Thank you so muchhh!!

**Comment by fernandaflx** [comment]
The error is now MUCH worse than before, not even the texts in the default language are appearing. As shown in the screenshots, I took the title of my form and the save button as examples. In both cases, all the languages enabled in my application are properly configured, but when running the app on the device, no text appears. Even after restarting the application, nothing changes, until the translations disappear, leaving only the default language.

I believe I'm not the only FlutterFlow user facing this unpleasant situation. I have a ready application, but it’s impossible to make it available to clients because of this error. And now, while developing a second app, I am starting to reconsider the use of FlutterFlow.

Please guys, find a way to fix this error as soon as possible. This issue has been open since February and nothing has changed so far.

<img width="889" alt="Captura de Tela 2024-09-12 às 18 04 56" src="https://github.com/user-attachments/assets/9ef0d25b-84e4-4dbd-990e-3fd44185edac">
<img width="889" alt="Captura de Tela 2024-09-12 às 18 05 05" src="https://github.com/user-attachments/assets/fdf1d5c9-b791-4409-8710-1d73c917413e">
![Captura de Tela 2024-09-12 à(s) 18 05 26](https://github.com/user-attachments/assets/5eb53d79-20c8-449c-ad28-52add6be8f5f)

**Comment by fernandaflx** [comment]
The error is now MUCH worse than before; not even the texts in the default language are showing up. As I show in the video, all the texts are properly configured, both for the default language and for the translations, but when running the app on the device, no text appears. Even after restarting the app, nothing changes, until the moment when the translations disappear, leaving only the default language.

I believe I’m not the only FlutterFlow user going through this unpleasant situation. I have a finished app, but it’s impossible to release it to clients due to this error. And now, while I’m developing another app, I’m starting to reconsider using FlutterFlow.

Please, guys, find a way to fix this error as soon as possible. This issue has been open since February, and nothing has changed so far.

https://github.com/user-attachments/assets/b39cd08b-4129-4d61-8144-af862a19fcfc

![screenshot](https://github.com/user-attachments/assets/b5bb2d9d-5e54-4e56-99fa-6c326157f453)

**Comment by Thesergiolg99** [comment]
how is this ticket on github closed?! This error is a BIG one, no solution has been delivered yet, everytime i copy any widget (no matter what widget, column, row etc) when pasting to other side, all translations breakes. We need help in here ASAP, because its so difficult to work like this.

**Comment by leighajarett** [comment]
Going to reopen as we investigate further - it's clear we must have missed something in our fix. I appreciate everyone's patience

**Comment by agreaves** [comment]
@fernandaflx could you please provide a Bug Report Code for your project / page? At the moment I'm unsure whether this is the same issue as the translations disappearing. Is the issue that the translations are there in the project, but not showing up in the app for you?

@Thesergiolg99 apologies for the issues you're facing! It was initially closed because we fixed the two main ways this happen, but it looks like we've missed a spot. Could you describe a bit more when it happens for you? Is it mainly happening when you are copy pasting widgets or in other cases? If you could also share a Bug Report Code that would be super helpful.

Thanks again, and hoping to fix this once and for all!

**Comment by Thesergiolg99** [comment]
> @fernandaflx could you please provide a Bug Report Code for your project / page? At the moment I'm unsure whether this is the same issue as the translations disappearing. Is the issue that the translations are there in the project, but not showing up in the app for you?
> 
> 
> 
> @Thesergiolg99 apologies for the issues you're facing! It was initially closed because we fixed the two main ways this happen, but it looks like we've missed a spot. Could you describe a bit more when it happens for you? Is it mainly happening when you are copy pasting widgets or in other cases? If you could also share a Bug Report Code that would be super helpful.
> 
> 
> 
> Thanks again, and hoping to fix this once and for all!

Thanks for your help.

Thats the point, it mainly happens when:

I have a page with content ( no matter what content). My app has english and spanish, being english the main language. When i copy the content of that page and paste ir anywhere in other pages or even in the same page, spanish translations disappear. Also it has happened sometimes that later on i navigate to any random page and suddenly a random translation of other pages are there.

Will try to get a report code.

Thanks in advance.

**Comment by ready4sea** [comment]
Agree with the description provided by @Thesergiolg99 :
- take a text widget with translations set in all languages
- duplicate the item
- it now only contains text in the main language, but all other translations have been lost
In addition, if you immediately try to set the translations, they most likely won't be saved. You need to go elsewhere and return to the duplicated text widget and only then set the translations.
Finally, I haven't encountered very often the last issue where a random translation will appear somewhere else. But it has happened.

**Comment by kazz91** [comment]
> @fernandaflx could you please provide a Bug Report Code for your project / page? At the moment I'm unsure whether this is the same issue as the translations disappearing. Is the issue that the translations are there in the project, but not showing up in the app for you?
> 
> @Thesergiolg99 apologies for the issues you're facing! It was initially closed because we fixed the two main ways this happen, but it looks like we've missed a spot. Could you describe a bit more when it happens for you? Is it mainly happening when you are copy pasting widgets or in other cases? If you could also share a Bug Report Code that would be super helpful.
> 
> Thanks again, and hoping to fix this once and for all!

I made a video now showing the bug (October 3rd, 2024):

https://github.com/user-attachments/assets/6d2df686-4257-4470-b812-e0d5357aca21

**Comment by MxMartin** [comment]
Are there any updates in this issue @agreaves ? I'd love to work with FlutterFlow again :)

**Comment by ready4sea** [comment]
Personally, I have noticed FF is much more stable now in this regard. The whole translation mechanism could be seriously enhanced. But at least, we're not losing data anymore.

**Comment by YassinElsherif** [comment]
I Agree with @ready4sea. It is getting much more stable now, even when you copy the element you don't lose data/translations.

**Comment by michael-mcroskey** [comment]
Thanks everyone for all the detailed reports and patience as we worked through this! We believe this issue has now been fixed and are closing it for now. If you’re still experiencing any translation-related problems, feel free to comment or open a new issue so we can dig in further. 🙏
